File: web/src/api/site-config.ts
import { request, type WebApiEnvelope } from './request'

export interface SiteConfigResponse {
  site_name: string
  logo_url: string
  login_captcha_enabled: boolean
  register_captcha_enabled: boolean
  password_reset_request_captcha_enabled: boolean
  password_reset_confirm_captcha_enabled: boolean
  real_name: RealNameConfig
}

export interface RealNameConfig {
  enabled: boolean
  required_for_order: boolean
  allowed_providers: string[]
  default_provider: string
  resubmit_enabled: boolean
  max_submit_attempts: number
  review_notice: string
}

export async function getSiteConfig() {
  const response = await request.get<WebApiEnvelope<SiteConfigResponse>>('/site-config')
  return response.data.data
}

File: server/internal/web/dto/site_config.go
package dto

/**
 * SiteConfigResponse 表示 Web 公开站点基础展示配置。
 */
type SiteConfigResponse struct {
	SiteName                           string `json:"site_name"`
	LogoURL                            string `json:"logo_url"`
	LoginCaptchaEnabled                bool   `json:"login_captcha_enabled"`
	RegisterCaptchaEnabled             bool   `json:"register_captcha_enabled"`
	PasswordResetRequestCaptchaEnabled bool   `json:"password_reset_request_captcha_enabled"`
	PasswordResetConfirmCaptchaEnabled bool   `json:"password_reset_confirm_captcha_enabled"`
}
